src/main/java/me/arminb/sara/dao/UserDAOImpl.java

```diff
@@ -115,17 +108,50 @@ public User findById(ObjectId id) throws DataAccessException {
                 return null;
             }
         } catch (MongoException e) {
-            logger.warn("Failed to search users on the database", e);
+            logger.warn("Failed to find users on the database", e);
             throw new DataAccessException();
         }
     }
 
+
     @Override
-    public boolean delete(ObjectId id) throws DataAccessException {
+    public User saveUser(User user) throws DataAccessException {
+        try{
+            MongoCollection<Document> collection = database.getCollection("users");
+
+            if (user.getId() == null){
+                user.setId(new ObjectId().toString());
+                user.setCreatedAt(new Date());
```

Comment:
ceratedAt doesn't need to be saved in the database. Also, use modified_at instead of modified_date